projectDocs/dev/developerGuide/developerGuide.md

```diff
@@ -105,68 +118,222 @@ Again, the English symbols are inherited by all other locales, so you need not i
+
+##### Example 3: The original gesture is defined to match a native shortcut {#TranslatingGesturesEx3}
+
+NVDA provides a script for Word document object named `toggleBold`.
+This script is mapped to the same gesture as the Word native shortcut to set text bold, i.e. `control+b` in English version of Word.
+However on French version of Word, the shortcut to turn text bold is `control+g`.
+The G stands for "gras" meaning "bold" in French.
+The following lines have been added in the French `gestures.ini` file to remap this script:
+
+```
+[NVDAObjects.window.winword.WordDocument]
+	None = kb:control+b, kb:control+[, kb:control+], "kb:control+shift+,", kb:control+shift+., kb:control+l, kb:control+r
+	toggleBold = kb:control+g, kb:control+shift+b
+```
+
+We can see that `control+b` has been unbound.
+This was necessary because it is the shortcut of another command in French version of Word.
+No remapping has been done for `toggleItalic` script since the shortcut is the same for French and English versions of Word.
+
+#### How to remap a shortcut key {#TranslatingGesturesSteps}
+
+##### Identify the class, the script and the original gesture to be remapped {#TranslatingGesturesStepIdentify}
+
+To edit the gesture.ini file, you will have to identify the class, the script and the original shortcut you want to remap.
+
+##### Case of a global command script {#TranslatingGesturesStepCaseGlobal}
+
+If the gesture to be remapped is a global command, you may execute the following steps to find out the class and the script name of the command:
+
+* activate input help (`NVDA+1`)
+* press the gesture you want to remap, e.g. `NVDA+]` (laptop layout)
+* de-activate input help (`NVDA+1`)
+* open the log (`NVDA+F1`)
+* find out the line corresponding to the moment you have executed the gesture, e.g.:
+  ```
+  Input help: gesture kb(laptop):NVDA+], bound to script rightMouseClick on globalCommands.GlobalCommands
+  ```
+
+The information you are searching is on this line:
+
+* script name: `rightMouseClick`
+* class name: `globalCommands.GlobalCommands` (Note that this is always this class for global commands)
+* original gesture: `kb(laptop):NVDA+]`
+
+##### Case of an application specific script {#TranslatingGesturesStepCaseApplication}
+
+In case you want to remap an application specific script, you will have to follow the same steps as those for a global command script.
+You just need to ensure before proceeding that you are in the targeted application.
+
+##### Case of an object specific script {#TranslatingGesturesStepCaseObject}
+
+Object specific scripts such as the ones linked to `NVDAObjects.window.winword.WordDocument` do not have generally a help message.
+So the previous technique will not work.
+You will need to explore NVDA's source code to find the class and the script name of the gesture you want to remap.

```

Comment:
It's not true that the previous technique will not work. Thus I suggest the following replacement:

```suggestion
For object specific scripts such as the ones linked to `NVDAObjects.window.winword.WordDocument`, you may follow the same steps as those for application specific script, paying attention to the two following points:
* You need to ensure before proceeding that the object to which the script is bound is focused.
* Some of these script have no help message, so you may not hear anything when executing them in input help mode; but the script's name and the class of the object will still appear in the log.

Though, the class  of the object appearing in the log may be a subclass of the one where the original gesture is actually bound. In this case, you will have to explore NVDA's source code to find this parent class.
```